### PR TITLE
WB-1142.4: Move `Popover` to Storybook

### PR DESCRIPTION
**Pull request by jandrade**
## Summary:
Last PR that migrates the `Popover` component to Storybook:

- Ported the existing Styleguidist examples.
- Included a new story to show how to use the `actions` prop.
- Included the `accessibility` info as well (using MDX).

Issue: WB-1142

Test plan:

Navigate to the Popover / Popover section and verify that all
the stories match with the React Styleguidist version:

Styleguidist: https://wonder-blocks.netlify.app/#section-popover

### CODE REVIEW

**Comment by jandrade** [comment]
The parent pull-request ([#1576](https://github.com/Khan/wonder-blocks/pull/1576)) has been merged into `main`, but this branch (`sb-popover4`) now has conflicts with the new base branch. These conflicts must be resolved before checks can complete on this pull-request.

**Review by jeresig — APPROVED** [review]
Great, thanks for porting this!

File: packages/wonder-blocks-popover/src/components/__docs__/popover.stories.js [hunk 1]
```diff
@@ -33,8 +58,276 @@ const styles = StyleSheet.create({
...
+    onClose: () => {},
+};
+
+/**
+ * Using a trigger element
+ */
+
+export const TriggerElement: StoryComponentType = Template.bind({});
+
+TriggerElement.args = {
```
[the remark this answers is not in the document]
Comment (nishasy) [reply packages/wonder-blocks-popover/src/components/__docs__/popover.stories.js:96]: This is actually the case with every example here except the Controlled story.

File: packages/wonder-blocks-popover/src/components/__docs__/popover.stories.js [hunk 2]
```diff
@@ -33,8 +58,276 @@ const styles = StyleSheet.create({
...
+                popover is opened. To see more details, please check the
+                **Accesibility section**.`,
+        },
+    },
+};
+
+/**
+ * With Actions
+ */
+export const WithActions: StoryComponentType = Template.bind({});
```
Comment (nishasy) [inline packages/wonder-blocks-popover/src/components/__docs__/popover.stories.js:224]: The "show code" section for the WithActions story doesn't show the `actions` prop in the code.

image

File: packages/wonder-blocks-popover/src/components/__docs__/popover.accessibility.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,78 @@
+import {Meta, Story, Canvas} from "@storybook/addon-docs";
```
Comment (nishasy) [inline packages/wonder-blocks-popover/src/components/__docs__/popover.accessibility.stories.mdx:1]: Yayy!! Accessibility section!

**Review by nishasy — COMMENTED** [review]
Personally, I use the "show code" sections a lot when I look at these examples, so I think it's important for them to show the full context of the stories.

File: packages/wonder-blocks-popover/src/components/__docs__/popover.stories.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (jandrade) [reply packages/wonder-blocks-popover/src/components/__docs__/popover.stories.js:96]: oh I think that it has to do with the `decorator` argument that is part of the default export. I'll take a look to see how I can improve that.

File: packages/wonder-blocks-popover/src/components/__docs__/popover.stories.js (hunk 2 above)
Comment (jandrade) [reply packages/wonder-blocks-popover/src/components/__docs__/popover.stories.js]: @nishasy I pushed a new commit with your suggestions. I did as much as I could to keep the snippets useful :). Let me know what you think.

https://5e1bf4b385e3fb0020b7073c-lbbnucutzz.chromatic.com/?path=/docs/popover-popover--with-actions

**Review by nishasy — APPROVED** [review]
Awesome! Thanks for your investigation into showing the code

**Comment by jandrade** [comment]
TODO: @nishasy and I discussed about adding an extra example to illustrate how the trigger element can be used in more complex scenarios. I'll do that in a separate PR.

**Pull request merged**